ext/mysqli/mysqli.c

```diff
@@ -1223,8 +1216,6 @@ void php_mysqli_fetch_into_hash(INTERNAL_FUNCTION_PARAMETERS, int override_flags
 				zval_ptr_dtor(&retval);
 			}
 			zend_fcall_info_args_clear(&fci, 1);
-		} else if (ctor_params) {
-			zend_throw_exception_ex(zend_ce_exception, 0, "Class %s does not have a constructor hence you cannot use ctor_params", ZSTR_VAL(ce->name));
```

Comment:
imho this error should remain, just not trigger for empty arrays.

Reply:
I shouldn't have tried to address the comments when busy... Sorry, I'll fix the issues soon.